sql/catalyst/src/main/scala/org/apache/spark/sql/catalog/v2/LookupCatalog.scala

```diff
@@ -62,6 +62,9 @@ trait LookupCatalog extends Logging {
     try {
       Some(lookupCatalog(SESSION_CATALOG_NAME))
     } catch {
+      case _: CatalogNotFoundException =>
+        logWarning("Session catalog is not defined")
+        None
```

Comment:
Could you print out the error message of `CatalogNotFoundException`?
Without that, previous error message is better for users because it gives the exact reason `Cannot find catalog plugin class for catalog 'session': xxx`.
```
$ bin/spark-shell --conf spark.sql.catalog.session=xxx
scala> spark.sessionState.analyzer.sessionCatalog
19/08/10 18:17:29 ERROR HiveSessionStateBuilder$$anon$1: Cannot load v2 session catalog
org.apache.spark.SparkException: Cannot find catalog plugin class for catalog 'session': xxx
```

Reply:
BTW, @jzhuge . Is this all places to give a warning instead of the stacks traces? 
With this PR, I still see the previous behavior.

Reply:
@dongjoon-hyun Thanks for the review. Your command line is not the case I tried to fix in the PR. In your case, the stack trace is helpful.

It seems that the current master has session catalog defined by default, so here is the command line to reproduce my case:
```
$ bin/spark-shell --master 'local[*]' --conf spark.sql.catalog.session=
...
Spark context available as 'sc' (master = local[*], app id = local-1565588237201).
Spark session available as 'spark'.
...
scala> spark.sessionState.analyzer.sessionCatalog
...
2019-08-11 22:37:24,216 ERROR [main] hive.HiveSessionStateBuilder$$anon$1 (Logging.scala:logError(94)) - Cannot load v2 session catalog
org.apache.spark.SparkException: Cannot find catalog plugin class for catalog 'session':
	at org.apache.spark.sql.catalog.v2.Catalogs.load(Catalogs.java:81)
...
res0: Option[org.apache.spark.sql.catalog.v2.CatalogPlugin] = None
```
Here is the stack trace does not add more information. And I am concerned that if any rule uses session catalog, we will see this long stack trace again and again.

Reply:
Could you fix both cases in this PR? I believe `typo` cases will be more frequent than the empty configuration cases.